### DISCUSSION TITLE
Japanese translation and some minor code changes

### DISCUSSION THREAD

**Discussion by NiyaShy** [NiyaShy/XB1ControllerBatteryIndicator#78] [Releases]
Not much new stuff in this either, "just" a new translation and a few code changes (including some updated packages relevant for building the binary/exe from code).

One rather small code change could be at least a bit relevant though: as reported in issue #72, there might have been a rare edge case that could cause the program to crash. As a precaution I've added a so-called try/catch function that should hopefully "catch" the error and prevent the crash. 

<hr /><em>This discussion was created from the release <a href='https://github.com/NiyaShy/XB1ControllerBatteryIndicator/releases/tag/v1.3.1.2'>Japanese translation and some minor code changes</a>.</em>

**Comment by jhay003** [comment]
It's working perfectly on me. 
additional comment instead of indicator better to add %.

**Reply by NiyaShy** [reply]
Do I understand it correctly that you're asking for percentage values instead of full/medium/low? As I've stated several times in different places in the past, the values you currently see are all I can get from the API, there is no percentage value to display.

**Reply by saulob** [reply]
Maybe you can add this info on the README file, I think it can help :)

**Reply by ghost** [reply]
I updated the firmware on my gamepad through xbox accessories and now the program does not display the charge of the gamepad. Writes that the gamepad is connected, getting the charge level and nothing more.

**Reply by NiyaShy** [reply]
A few more details would be nice. Controller model (printed on the sticker in the battery compartment), are you connected via proprietary dongle or bluetooth, and do you mean the "waiting for data" state?

**Reply by NiyaShy** [reply]
Just tried it with my own Series X/S controller (model 1914), updated the firmware from 5.13.3143.0 to 5.17.3202.0, and battery level detection still works fine via the proprietary (old/chunky) dongle.

**Reply by ghost** [reply]
![photo_2023-08-09_15-20-43](https://github.com/NiyaShy/XB1ControllerBatteryIndicator/assets/141753184/068450a4-39cb-4d64-a4f1-e6109e26b365)
Controller xbox one model 1708. I connect to the laptop via bluetooth. Recently, I needed to remap the buttons on the gamepad and in the Microsoft store I downloaded the "xbox accessories" application and after launching it, it suggested that I first update the gamepad firmware and only after that I was able to enter the application. But now the question mark is on the indicator and writes that "Controller One found but still waiting for battery data"
firmware vervion 5.17.3202.0

**Reply by NiyaShy** [reply]
And there you (probably) have your answer: Bluetooth. As mentioned on the issues section of the main page, the new series X/S controllers have a 99,9% chance of reporting no or complete garbage battery levels via BT, and while I got reports that older revisions that already supported BT (XBOne S, not series S) worked fine over it, newer reports all said that it doesn't work any more. So it seems like the newer firmware revisions changed something that screws up the reporting... You can find a long discussion about it in #49, and the only real "fix" is to use a proprietary dongle and not BT. Though it doesn't seem to matter if you use an official MS dongle or a Chinese knock-off version.

**Comment by ArthurF2023** [comment]
Good morning

On mine here it is giving this message with the question mark on the indicator and it says "Controller One found, but still waiting for battery data" so have an idea of ​​getting the data from this screen here because on yours there is a low battery notification maybe this solve the problem with the latest control firmware update...

I used another one to test his name and"XboxWirelessControllerBatteryLevels"
Link: https://github.com/nymwo/XboxWirelessControllerBatteryLevels
As it shows battery level but there is no notification
![1](https://github.com/NiyaShy/XB1ControllerBatteryIndicator/assets/130934983/b77c4090-9b40-45c5-8e7a-026884a0831c)

**Reply by NiyaShy** [reply]
Hey, I just tried that other tool, and for me it claimed that it couldn't find my controller. Combined with your screenshot (and the % values shown there) I guess you connected the controller via bluetooth? If so, please see my previous comments here and/or issue #49 for details why it's not working.
After looking at the code of the other tool it's good to know that BT controllers apparently can be polled via powershell commands. That would be way easier than finding a compatible library, so there could still be hope that xb1cbi one day (also) supports BT controllers.

For now, your only chance to get it working would be to use one of Microsofts proprietary dongles (or a knock-off version).

**Reply by jasonxbourne** [reply]
> Hey, I just tried that other tool, and for me it claimed that it couldn't find my controller. Combined with your screenshot (and the % values shown there) I guess you connected the controller via bluetooth? If so, please see my previous comments here and/or issue #49 for details why it's not working. After looking at the code of the other tool it's good to know that BT controllers apparently can be polled via powershell commands. That would be way easier than finding a compatible library, so there could still be hope that xb1cbi one day (also) supports BT controllers.
> 
> For now, your only chance to get it working would be to use one of Microsofts proprietary dongles (or a knock-off version).

That's weird... It WAS working with BT for years, now suddenly it stop working. Anyway I plug my Microsoft Device Controller and now it shows the %.
